recipe/meta.yaml

```
@@ -58,7 +58,8 @@ test:
     - test -f $PREFIX/etc/cni/net.d/00_bridge.conflist
 
     # UserNS test
-    - singularity build --sandbox lolcow docker://godlovedc/lolcow  # [not aarch64]
+    # ngam: disabling this for ppc64le for now (fails with mount error)
+    - singularity build --sandbox lolcow docker://godlovedc/lolcow  # [linux64]
```

Assuming the CI now passes can we try running this for the other architechtures:
```suggestion
    - singularity build --sandbox lolcow docker://godlovedc/lolcow
```
I have no idea if it'll work for the emulated aarch64 build but I'd like to check rather that deploy a broken build if we can 🤞 (I've deployed a lot of unrelocatable singularity builds before these tests were added 😅 )

---

Yes, let me try to enable this again now that we know everything works fine on linux_64

---

@chrisburr, using `--sandbox` here may not be that useful imo on an HPC. I never had access to singularity where I also could build locally (so usually I would use the remote builder). However, I often want to use `--sandbox` with the remote builder so that I can use `--writable` when running the container. Do you think `--sandbox` is important for the latter use case when using a remote builder? Do we need to add a test for `--writable` as well somewhere? Generally, should we somehow test the remote builder in this or are we covered?

---

> using --sandbox here may not be that useful imo on an HPC

Honestly I don't even know what the `--sandbox` option does 😄 My usage of singularity is almost entirely with already unpacked container images from CVMFS.

The solution in eceeb65d7b50710a6719f93f1e1d0dc05b501c63 seems nice.

> Generally, should we somehow test the remote builder in this or are we covered?

I don't know much about how the remote builders work but I doubt it's worth testing them from a packaging perspective. If someone finds a packaging bug with it we can re-evaluate.

---

Okay, here's what I know about the `--sandbox` flag from actual use/experience:
- basically you need it when you want to modify the content of a container (i.e. have write access to the content) which sometimes is necessary if you're compiling or making plots, etc.; however, I really think it is a bad design idea if you design a container and the user has to have write access to its directories. The proper way to do it is direct all the write to the user's own directories that are mounted (which is done automatically with singularity anyway)
- you build with `--sandbox` and then you run with `--writable`, otherwise the whole `--sandbox` flag is useless

One additional consideration: I believe this test won't do us many favors here because unless the HPC admins have set up the file system correctly for Singularity , the `--sandbox` flag will likely fail regardless to what we do here. And, honestly, if HPC admins have gone through the work to set up everything correctly for these things, they would have installed Singularity as well.

Finally, to reiterate, I would be quite surprised if HPC admins allow users to build locally. Usually building requires sudo or some sort of fakeroot, which most admins would likely not tolerate. Because I think the remote builder will likely be the only option for the `--sandbox` operation, I will think a little about the remote builder and do some testing once we have this out to see if we need to adjust anything.

---

> CVMFS

! Didn't know you were from CERN! I am still technically part of one of the low-key experiments and still have access to random infrastructure and I got the stupid cernbox reply-all drama last week ... 👀